ydb/library/yql/providers/s3/provider/yql_s3_io_discovery.cpp

```diff
@@ -92,6 +92,7 @@ class TS3IODiscoveryTransformer : public TGraphTransformerBase {
               State_->Configuration->MinDesiredDirectoriesOfFilesPerQuery,
               State_->Configuration->MaxInflightListsPerQuery,
               State_->Configuration->AllowLocalFiles)) {
+        State_->Gateway = gateway;
```

Comment:
If we need Gateway in the state, it is better to assign it in InitS3Provider
https://github.com/ydb-platform/ydb/blob/main/ydb/core/kqp/host/kqp_host.cpp#L1470
and do not pass as param over dozen constructors.